local-cluster/src/cluster_tests.rs

```diff
@@ -678,3 +689,23 @@ pub fn submit_vote_to_cluster_gossip(
         socket_addr_space,
     )
 }
+
+pub fn new_tpu_quic_client(
+    contact_info: &ContactInfo,
+    connection_cache: Arc<ConnectionCache>,
+) -> Result<QuicTpuClient, TpuSenderError> {
+    let rpc_pubsub_url = format!("ws://{}/", contact_info.rpc_pubsub().unwrap());
+    let rpc_url = format!("http://{}", contact_info.rpc().unwrap());
+
+    let cache = match &*connection_cache {
+        ConnectionCache::Quic(cache) => cache,
+        ConnectionCache::Udp(_) => panic!("Expected a Quic ConnectionCache. Got UDP"),
+    };
+
+    TpuClient::new_with_connection_cache(
+        Arc::new(RpcClient::new(rpc_url)),
+        rpc_pubsub_url.as_str(),
+        TpuClientConfig::default(),
+        cache.clone(),
```

Comment:
ya it is because `cache` is of type `&Arc<ConnectionCache>` and `new_with_connection_cache()` requires ownership over the `Arc` aka `Arc<ConnectionCache>`. If I try to match instead on `connection_cache` instead of `&*connection_cache`, then I end up matching on the `Arc` instead of the underlying `connection_cache`